bookmarks/models.py

```diff
@@ -7,11 +7,31 @@
 
 
 class Bookmark (models.Model):
-    user = models.ForeignKey(User, on_delete=models.CASCADE)
-    course = models.ForeignKey(Course, on_delete=models.CASCADE, default='SOME STRING')
+    user = models.ForeignKey(User, on_delete=models.PROTECT)
+    course = models.ForeignKey(Course, on_delete=models.PROTECT)
     url = models.URLField(max_length=200)
     urlname = models.CharField(max_length=30)
-    last_view_date = models.DateTimeField(auto_now=True, null=True)
+    last_view_date = models.DateTimeField(auto_now=True, blank=True)
 
     def __str__(self):
         return self.urlname
+
+    @classmethod
+    def add_bookmark(cls, user, course, url, urlname):
+        """Creates new bookmark"""
+        b = Bookmark.objects.create(user=user, course=course, url=url, urlname=urlname)
+        b.save()
+        pass
+
+    def remove_bookmark(cls):
+        '''deleted bookmark'''
+        cls.delete()
```

Comment:
Don't you need to pass in an ID, get the object and then call delete?

Reply:
If you are getting the bookmark first and then passing it in to this function, which is what you're doing in the test, (which is kind of pointless), you would want to have another parameter.
```
def remove_bookmark(cls, bm):
    bm.delete()
```
The first cls parameter passes in the class itself. The parameter you pass in afterwards is the object .

Reply:
Fixed it and I agree. I'll try to think about more interesting functions that I can run tests on.